common/display/displayplanemanager.cpp

```diff
@@ -458,6 +460,39 @@ void DisplayPlaneManager::ValidateCachedLayers(
   *render_layers = needs_gpu_composition;
 }
 
+bool DisplayPlaneManager::ValidateSurfaceDamage(
+    const std::vector<OverlayLayer> *layers,
+    const std::vector<OverlayLayer> *previous_layers) {
+  auto layer_begin = layers->begin();
+  auto layer_end = layers->end();
+  bool validate = true;
+  size_t num_layers = layers->size();
+
+  for (size_t i = 0; i <= num_layers; i++) {
+    const HwcRegion &current_surface_damage = layers->at(i).GetSurfaceDamage();
+
+    if (current_surface_damage.kNumRects == 1) {
+      const HwcRect<int> *rect = current_surface_damage.kRects;
```

Comment:
Instead of this I think what we should do is just check if kNumRects =0, if so than assume buffer is same as previous. If it's not same for now we can fallback to 3D (later can think if using 2d block makes sense).